### PR TITLE
feat/products-endpoint

### PR DESCRIPTION
**Pull request by Esther-Guimaraes**

### CODE REVIEW

File: src/utils/mockApi.ts [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
 import phones from '../../public/api/phones.json';
 import tablets from '../../public/api/tablets.json';
 import accessories from '../../public/api/accessories.json';
-import { ProductType } from '../types/ProductType';
 import { ProductDetails } from '../types/ProductDetails';
+import { ProductsResponse } from '../types/ProductsResponse';
+import axios from 'axios';
 
 const productsAll = [...tablets, ...phones, ...accessories];
+const API_URL = 'http://localhost:3005';
```
Comment (eduvinagre) [inline src/utils/mockApi.ts:9]: vamos colocar isso num arquivo .env, por favor, pq na produção eu vou precisar colocar o endereço da api de verdade

**Review by eduvinagre — CHANGES_REQUESTED** [review]
Very small fix to be done.

File: src/utils/mockApi.ts (hunk 1 above)
Comment (Esther-Guimaraes) [reply src/utils/mockApi.ts:9]: pensei isso também, vou colocar

**Review by eduvinagre — APPROVED** [review]
Great! Thank you!

File: src/pages/phones-page/PhonesPage.tsx [hunk 2]
```diff
@@ -50,28 +50,22 @@ const PhonesPage = () => {
...
   useEffect(() => {
     setSkeleton(true);
-
-    fetchProducts()
-      .then(data => {
-        setPhones(data.filter((product: ProductType) => product.category === 'phones'));
+    fetchProducts('phones', currentPage, pageSize)
+      .then(res => {
+        setTotalPages(res.data.totalPages);
+        setPhones(res.data.products);
```
Comment (eduvinagre) [inline src/pages/phones-page/PhonesPage.tsx:63]: So I think you are missing the use of axios here and on the following parts.

File: src/pages/tablets-page/TabletsPage.tsx [hunk 3]
```diff
@@ -50,28 +50,22 @@ const TabletsPage = () => {
...
   useEffect(() => {
     setSkeleton(true);
-
-    fetchProducts()
-      .then(data => {
-        setTablets(data.filter((product: ProductType) => product.category === 'tablets'));
+    fetchProducts('tablets', currentPage, pageSize)
+      .then(res => {
+        setTotalPages(res.data.totalPages);
+        setTablets(res.data.products);
```
Comment (eduvinagre) [inline src/pages/tablets-page/TabletsPage.tsx:63]: Here

File: src/utils/mockApi.ts [hunk 4]
```diff
@@ -1,17 +1,20 @@
...
-    setTimeout(() => {
-      resolve(products as ProductType[]);
-    }, 500);
+export const fetchProducts = async (category: string, page: number, limit: number) => {
+  return await axios.get<ProductsResponse>(`${API_URL}/products/`, {
+    params: {
+      category,
+      page,
+      limit,
+    },
```
Comment (eduvinagre) [inline src/utils/mockApi.ts]: 🤔  Why not use in the others?

File: src/pages/accessories-page/AccessoriesPage.tsx [hunk 5]
```diff
@@ -50,28 +50,22 @@ const AccessoriesPage = () => {
...
   useEffect(() => {
     setSkeleton(true);
-
-    fetchProducts()
-      .then(data => {
-        setAccessories(data.filter((product: ProductType) => product.category === 'accessories'));
+    fetchProducts('accessories', currentPage, pageSize)
+      .then(res => {
+        setTotalPages(res.data.totalPages);
+        setAccessories(res.data.products);
```
Comment (eduvinagre) [inline src/pages/accessories-page/AccessoriesPage.tsx:63]: another one missing the use of axios I think

**Review by eduvinagre — COMMENTED** [review]
Great job!! Keep it up! But I think you can improve the use of axios overall

**Pull request merged**